### PR TITLE
[FLINK-28792] Upgrade to JOSDK 3.1.0

### PR DESCRIPTION
**Pull request by gyfora**
## What is the purpose of the change

Upgrade josdk to 3.1.0

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: no
  - Core observer or reconciler logic that is regularly executed: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by gyfora** [comment]
cc @morhidi

File: docs/layouts/shortcodes/generated/kubernetes_operator_config_configuration.html [hunk 1]
```diff
@@ -112,7 +112,7 @@
         </tr>
         <tr>
             <td><h5>kubernetes.operator.reconcile.parallelism</h5></td>
-            <td style="word-wrap: break-word;">5</td>
+            <td style="word-wrap: break-word;">10</td>
```
Comment (morhidi) [inline docs/layouts/shortcodes/generated/kubernetes_operator_config_configuration.html:115]: why do we need to increase the parallelism?

Comment (gyfora) [reply docs/layouts/shortcodes/generated/kubernetes_operator_config_configuration.html:115]: We use the default from the josdk we don’t set this value explicitly anywhere

Comment (morhidi) [reply docs/layouts/shortcodes/generated/kubernetes_operator_config_configuration.html:115]: makes sense, didn't notice it

**Comment by morhidi** [comment]
+1 for merging

**Pull request merged**